### DISCUSSION TITLE
Follow Up - save links for future consumption

### DISCUSSION THREAD

**Discussion by lannonbr** [johnlindquist/kit#910] [Share]
I frequently want to find ways of saving links that I consume across devices that I want to come back to later. With this, I tried out the new `fields` form available in the Sept 2022 version of Script Kit in which I grab the current tab with `getActiveTab` and auto fill a link field and then I just have to fill in a name and comment field and then I save it off to AWS DynamoDB.

Following up with this (no pun intended), I'm thinking of ways of passing the content back to me at a convenient time with ideas like:

- weekly email on a Sunday / Monday morning to see what I saved that week
- discord bot for me to fetch a random link
- etc

GitHub Repo: https://github.com/lannonbr/follow-up
Save link script: https://github.com/lannonbr/follow-up/blob/main/functions/save-a-link.js

**Comment by lannonbr** [comment]
I originally started out with the local `db` functionality Script Kit provides, but I wanted it to save across multiple devices and I was starting to overengineer syncing functionality of pushing the data up once a week, but the devices I use this on are almost always connected to the internet so sending it to a remote DB immediately I feel was a good way of not going too deep down a rabbit hole.

**Comment by johnlindquist** [comment]
Awesome idea:

Just riffing a bit, the "googlethis" lib could pull the description from the first google result:

> 📝  I've also been using `template` more than `fields` recently. It's a just a personal preference

```js
// Name: URL Saver

import "@johnlindquist/kit"

let url = await getActiveTab()

let google = await npm("googlethis")
let response = await google.search(url)

let { title, description } = response.results[0]

let content = await template(`
[${title}](${url})

${description}

\${1:notes}

\${2:tags}
`.trim())

// do something with content
await div(md(content))
```

[Open testing-url-with-description in Script Kit](https://scriptkit.com/api/new?name=testing-url-with-description&url=https://gist.githubusercontent.com/johnlindquist/9c9b0da9dd7ce9df46f603a070bf1226/raw/9a8dbb41f7dbeb3ac2db2b8158ecf29be352d1aa/testing-url-with-description.ts")

Another thought, you can reach into Chrome and pull anything you want from the page like in this example. OpenGraph image maybe? 🤔

https://github.com/johnlindquist/kit/discussions/695
